specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json

```diff
@@ -0,0 +1,363 @@
+          ],
+          "operationId": "Baselines_List",
+          "description": "**Lists the metric baseline values for a resource**.",
+          "x-ms-examples": {
+            "Get metric baselines": {
+              "$ref": "./examples/metricBaselines.json"
+            }
+          },
+          "parameters": [
+            {
+              "$ref": "#/parameters/ResourceUriParameter"
+            },
+            {
+              "$ref": "#/parameters/MetricNamesParameter"
+            },
+            {
+              "$ref": "#/parameters/MetricNamespaceParameter"
+            },
+            {
+              "$ref": "#/parameters/TimespanParameter"
+            },
+            {
+              "$ref": "#/parameters/IntervalParameter"
+            },
+            {
+              "$ref": "#/parameters/AggregationsParameter"
+            },
+            {
+              "$ref": "#/parameters/SensitivitiesParameter"
+            },
+            {
+              "name": "$filter",
+              "in": "query",
+              "type": "string",
+              "description": "The **$filter** is used to reduce the set of metric data returned.<br>Example:<br>Metric contains metadata A, B and C.<br>- Return all time series of C where A = a1 and B = b1 or b2<br>**$filter=A eq ‘a1’ and B eq ‘b1’ or B eq ‘b2’ and C eq ‘*’**<br>- Invalid variant:<br>**$filter=A eq ‘a1’ and B eq ‘b1’ and C eq ‘*’ or B = ‘b2’**<br>This is invalid because the logical or operator cannot separate two different metadata names.<br>- Return all time series where A = a1, B = b1 and C = c1:<br>**$filter=A eq ‘a1’ and B eq ‘b1’ and C eq ‘c1’**<br>- Return all time series where A = a1<br>**$filter=A eq ‘a1’ and B eq ‘*’ and C eq ‘*’**.",
+              "required": false
+            },
+            {
+              "$ref": "#/parameters/BaselineResultTypeParameter"
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            }
+          ],
+          "responses": {
+            "default": {
+              "description": "Error response describing why the operation failed.",
+              "schema": {
+                "$ref": "#/definitions/ErrorResponse"
+              }
+            },
+            "200": {
+              "description": "Successful request to get the list of metric baseline values.",
+              "schema": {
+                "$ref": "#/definitions/MetricBaselinesResponse"
+              }
+            }
+          }
+        }
+      }
+    },
+    "definitions": {
+      "ErrorResponse": {
+        "description": "Describes the format of Error response.",
+        "type": "object",
+        "properties": {
+          "code": {
+            "description": "Error code",
+            "type": "string"
+          },
+          "message": {
+            "description": "Error message indicating why the operation failed.",
+            "type": "string"
+          }
+        }
+      },
+      "MetricBaselinesResponse": {
```

Comment:
I understand that this is trying to align to /metrics API, but this response violates the Resource Provider contract in its current form. 
1) The URI: /metricBaselines implies a collection and should return a collection response (value: [{....}]).  If it is a true singleton, you can use /metricBaselines/default, but the collection path should return the collection response in that case as well.
2) The response is not well formed, and should have the format:
`{"id":"<id>",
"type":"<type">,
"name":"<name>",
"properties":{}
}`

In effect, most of this should be pushed into the "properties" object. 

While this deviates from the response for /metrics, it achieves contractual compliance, enables response authorization checks, and establishes consistency with the remainder of APIs.

Reply:
You can use references to the common proxy resource model in https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json

Reply:
Ok, this makes sense. While seperating this API even more from the /metrics API, the common structure makes sense. I've updated the swagger to return a collection (only value[{}]) and each item in the correct format as above